File: IRemember/app/src/main/java/ru/goodibunakov/iremember/fragment/SplashFragment.java
package ru.goodibunakov.iremember.fragment;


import android.os.AsyncTask;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;

import java.util.Objects;

import ru.goodibunakov.iremember.R;


public class SplashFragment extends Fragment implements Animation.AnimationListener {

    private Animation splash_in;
    private Animation splash_out;

    public SplashFragment() {
        // Required empty public constructor
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {

        new SplashTask().execute();

        splash_in = AnimationUtils.loadAnimation(getActivity(), R.anim.splash_anim_in);
        splash_in.setAnimationListener(this);
        container.startAnimation(splash_in);
        splash_out = AnimationUtils.loadAnimation(getActivity(), R.anim.splash_anim_out);
        splash_out.setAnimationListener(this);
        splash_out.setStartOffset(1000);

        return inflater.inflate(R.layout.fragment_splash, container, false);
    }


    @Override
    public void onAnimationStart(Animation animation) {
    }

    @Override
    public void onAnimationEnd(Animation animation) {
        if (animation == splash_in) {
            Objects.requireNonNull(getView()).startAnimation(splash_out);
        }
        if (animation == splash_out) {
            if (getActivity() != null) {
                getActivity().getSupportFragmentManager()
                        .beginTransaction()
                        .remove(SplashFragment.this)
                        .commit();
            }
        }
    }

    @Override
    public void onAnimationRepeat(Animation animation) {
    }


    private static class SplashTask extends AsyncTask<Void, Void, Void> {

        SplashTask() {
        }

        @Override
        protected Void doInBackground(Void... voids) {

            try {
                Thread.sleep(1500);
            } catch (InterruptedException e) {
                e.printStackTrace();
            }
            return null;
        }
    }
}

File: IRemember/app/src/main/java/ru/goodibunakov/iremember/presentation/view/fragment/SplashFragment.kt
package ru.goodibunakov.iremember.presentation.view.fragment

import android.os.Bundle
import android.view.View
import android.view.animation.Animation
import android.view.animation.AnimationUtils
import by.kirich1409.viewbindingdelegate.viewBinding
import moxy.MvpAppCompatFragment
import moxy.presenter.InjectPresenter
import moxy.presenter.ProvidePresenter
import ru.goodibunakov.iremember.BuildConfig
import ru.goodibunakov.iremember.R
import ru.goodibunakov.iremember.databinding.FragmentSplashBinding
import ru.goodibunakov.iremember.presentation.presenter.SplashFragmentPresenter

class SplashFragment : MvpAppCompatFragment(R.layout.fragment_splash), SplashFragmentVew,
    MyAnimationListener {

    private var splashIn: Animation? = null
    private var splashOut: Animation? = null

    private val binding by viewBinding(FragmentSplashBinding::bind)

    @InjectPresenter
    lateinit var splashFragmentPresenter: SplashFragmentPresenter

    @ProvidePresenter
    fun providePresenter(): SplashFragmentPresenter {
        return SplashFragmentPresenter()
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        binding.version.text =
            String.format(resources.getString(R.string.version), BuildConfig.VERSION_NAME)

        splashIn = AnimationUtils.loadAnimation(activity, R.anim.splash_anim_in)
        splashOut = AnimationUtils.loadAnimation(activity, R.anim.splash_anim_out)

        splashIn?.setAnimationListener(this)
        splashOut?.setAnimationListener(this)

        binding.root.startAnimation(splashIn)
        splashOut?.startOffset = 1000
    }

    override fun close() {
        if (activity != null) {
            requireActivity().supportFragmentManager
                .beginTransaction()
                .remove(this@SplashFragment)
                .commitAllowingStateLoss()
        }
    }

    override fun onAnimationEnd(animation: Animation) {
        if (animation === splashIn) {
            view?.startAnimation(splashOut)
        }
        if (animation === splashOut) {
            splashFragmentPresenter.haveToClose()
        }
    }
}